lib/api/core/indexCache.js

```diff
@@ -48,58 +49,48 @@ class IndexCache {
   }
 
   init () {
-    const engine = this.kuzzle.internalEngine;
-    const kuzzleInfoFieldsRegex = new RegExp(`^_kuzzle_info\\.(${Object.keys(this.commonMapping._kuzzle_info.properties).join('|')})$`);
+    const internalEngine = this.kuzzle.internalEngine;
 
-    return engine.listIndexes()
+    return internalEngine.listIndexes()
       .then(indexes => indexes.filter(index => index !== this.kuzzle.internalEngine.index))
       .then(indexes => {
         const promises = [];
 
-        for (const index of indexes.filter(idx => idx !== this.kuzzle.internalEngine.index)) {
+        for (const index of indexes) {
           this.indexes[index] = [];
           this.defaultMappings[index] = _.cloneDeep(this.commonMapping);
 
-          promises.push(engine.getFieldMapping({
-            index,
-            fields: '_kuzzle_info.*'
-          })
-            .then(kuzInfoMappings => {
-              if (kuzInfoMappings[index] && kuzInfoMappings[index].mappings) {
-                for (const collection of Object.keys(kuzInfoMappings[index].mappings)) {
-                  const collectionInfo = kuzInfoMappings[index].mappings[collection];
-
-                  for (const dotField of Object.keys(collectionInfo).filter(f => kuzzleInfoFieldsRegex.test(f))) {
-                    Object.assign(this.defaultMappings[index]._kuzzle_info.properties, collectionInfo[dotField].mapping);
-                  }
-                  // do not break to catch cases where not all fields are defined
-                }
-              }
-
-              return engine.listCollections(index);
-            })
-            .then(collections => {
-              const updateMappingPromises = [];
+          promises.push(
+            internalEngine.listCollections(index)
+              .then(collections => {
+                const promisesAgain = [];
 
-              this.indexes[index] = collections;
+                for (const collection of collections) {
+                  this.indexes[index].push(collection);
 
-              for (const collection of collections) {
-                updateMappingPromises.push(engine.updateMapping(collection, {
-                  properties: this.defaultMappings[index]
-                }, index));
-              }
+                  promisesAgain.push(
+                    this.kuzzle.internalEngine.applyDefaultMapping(index, collection, this.defaultMappings[index])
+                  );
+                }
 
-              return Bluebird.all(updateMappingPromises);
-            })
+                return Promise.all(promisesAgain);
+              })
+              .then(defaultMappings => {
+                for (const defaultMapping of defaultMappings) {
+                  this.defaultMappings[index] = defaultMapping;
```

Comment:
I don't understand this loop. You're assigning the `defaultMapping`s returned by the subsequent calls to `this.kuzzle.internalEngine.applyDefaultMapping` to the same `this.defaultMappings[index]`. So, only the last one gets assigned to `this.defaultMappings[index]`.
Or, what am I missing?

Reply:
Actually they are all the same since we are working on the same index, I can just keep one assignation.